tests/test_packages/test_protocols/test_fipa.py

```diff
@@ -290,16 +288,217 @@ def test_fipa_decoding_unknown_performative():
+                                                                          dialogue_reference=cfp_msg.dialogue_reference,
+                                                                          is_seller=True)
+
+        # Register the dialogue to the dictionary of dialogues.
+        self.seller_dialogues.dialogues[seller_dialogue.dialogue_label] = cast(FIPADialogue, seller_dialogue)
+
+        # Extend the incoming list of messages.
+        seller_dialogue.incoming_extend(cfp_msg)
+
+        # Check that both fields in the dialogue_reference are set.
+        last_msg = seller_dialogue.last_incoming_message
+        dialogue_reference = cast(Tuple[str, str], last_msg.body.get("dialogue_reference"))
+        assert dialogue_reference[0] != "" and dialogue_reference[1] == ""
+
+        # Checks if the message we received is permitted for a new dialogue or if it is a registered dialogue.
+        assert self.seller_dialogues.is_permitted_for_new_dialogue(seller_dialogue.last_incoming_message), \
+            "Should be permitted since the first incoming msg is CFP"
+
+        # Generate a proposal message to send to the client.
+        proposal = [
+            Description({"foo1": 1, "bar1": 2}),
+            Description({"foo2": 1, "bar2": 2}),
+        ]
+        message_id = cfp_msg.message_id + 1
+        target = cfp_msg.message_id
+        proposal_msg = FIPAMessage(message_id=message_id,
+                                   dialogue_reference=seller_dialogue.dialogue_label.dialogue_reference,
+                                   target=target,
+                                   performative=FIPAMessage.Performative.PROPOSE,
+                                   proposal=proposal)
+        proposal_msg.counterparty = "seller"
+
+        # Extends the outgoing list of messages.
+        seller_dialogue.outgoing_extend(proposal_msg)
+
+        # Client received the message and we extend the incoming messages list.
+        client_dialogue.incoming_extend(proposal_msg)
+
+        # Check that both fields in the dialogue_reference are set.
+        last_msg = client_dialogue.last_incoming_message
+        dialogue_reference = cast(Tuple[str, str], last_msg.body.get("dialogue_reference"))
+        assert dialogue_reference[0] != "" and dialogue_reference[1] != ""
+
+        assert not self.client_dialogues.is_permitted_for_new_dialogue(client_dialogue.last_incoming_message),\
+            "Should not be permitted since we registered the cfp message."
+
+        response = self.client_dialogues.is_belonging_to_registered_dialogue(proposal_msg, agent_addr="client")
+        assert response, "We expect the response from the function to be true."
+
+        # Retrieve the dialogue based on the received message.
+        retrieved_dialogue = self.client_dialogues.get_dialogue(proposal_msg, "client")
+        assert retrieved_dialogue.dialogue_label in self.client_dialogues.dialogues
```

Comment:
this is a somewhat pointless assertion, you would not be able to retrieve the dialogue above , so more sensible to check that the retrieved dialogue is not None